src/cdk-experimental/menu/context-menu.ts

```diff
@@ -57,10 +57,13 @@ export class ContextMenuTracker {
 /** Configuration options passed to the context menu. */
 export type ContextMenuOptions = {
   /** The opened menus X coordinate offset from the triggering position. */
-  offsetX: number;
+  offsetX?: number;
 
   /** The opened menus Y coordinate offset from the triggering position. */
-  offsetY: number;
+  offsetY?: number;
+
+  /** Ordered list of preferred positions, from most to least desirable. */
+  preferredPositions?: ConnectedPosition[];
```

Comment:
There was a TODO saying we should expose this, but do we expect people to actually want to change this? Even the offset seems like something people probably would need to customize. Should we just remove this whole options object and add it later if people ask for it?

Reply:
I still can't really think of any reason why someone would want to use something other than the default, but I guess we want our CDK APIs to be flexible. I'll add something more like what we have for mat-menu. Actually looking at the mat-menu one brings up another thing: should we have optional overlays for the CDK menu? Right now it just seems to not use an overlay

Reply:
> Right now it just seems to not use an overlay

I don't follow this point. It seems to be using overlays as far as I can tell.

Reply:
oh wrong term 🤦  I meant backdrop

Reply:
We could try using the outside click dispatcher as an alternative to the backdrop for capturing outside clicks.

Reply:
It already seems to handle outside clicks fine, but I was more thinking in case someone wants to "gray out" the page with a semi-transparent overlay (hence why I accidentally called it an overlay)

Reply:
IMO that doesn't make sense for a menu.

Reply:
I would probably lean towards not including a backdrop and then adding it if there's enough demand

Reply:
Oh, and for the positioning: we might want to do a quick survey of menus in the wild to see if any have a really good reason to deviate from what I'd consider "standard dropdown positioning". Some off the top of my head thoughts:
* I could imagine a menu on mobile with positioning that sets `middle` + `center` for both so that it appears over the trigger
* I could imagine wanting to center-align menus either above or below the trigger
* I could imagine menus where the trigger is in something like a sidenav or side panel that would want to position themselves horizontally next to the trigger rather than above or below

Reply:
Looking around I'll also add:
- standard but with custom offset (e.g. gap between trigger and menu, or slightly overlapping trigger)
- standard but completely covering the trigger (e.g. old material select, chrome native select)

Maybe it makes sense to just take in a custom `FlexibleConnectedPositionStrategy`? That would give users the most control and if a particular app wants to create a more constrained position builder API for their particular library they're welcome to. I don't think it really makes sense to include such an API at this level.

Reply:
Oh it looks like that's not super easy to do since the `FlexibleConnectedPositionStrategy` requires a bunch of injected things to be passed to its constructor. In that case I propose either:
1. [Preferred] Just keep it how I have it here. Forward offsets, preferred positions, maybe other things if people request them to the position strategy
2. Allow users to specify a `customPosition` function that receives the `FlexibleConnectedPositionStrategy`, updates it, and returns the new strategy

Reply:
I added a demo for the custom config to the dev app. I noticed that only the context menu trigger lets you set a global custom config, the menu item trigger does not. I think this is because the menu item trigger has 3 possible cases:
- no parent menu
- horizontal parent menu
- vertical parent menu

People would probably want to be able to set a global default per context, since a single position strategy wouldn't make sense for all of this. Should I create a global config with separate settings for each of these? Or should we maybe not do the global config approach and instead just have a directive `[cdkMenuPosition]="..."` that allows users to set it for a specific scope?

Reply:
I discussed with @jelbourn offline and we decided it made the most sense to just have an `@Input()` for position on the trigger directives instead